### PR TITLE
Close Dax89/harbour-sailorgram#53

### PR DESCRIPTION
**Pull request by mentaljam**
Add ability to the show/hide search field on the DialogsPage (by default is hidden)

### CODE REVIEW

**Comment by Dax89** [comment]
There is an easier way to do that, with less code!

First of all, this boolean:

```
property bool showSearchField: false
```

It's better to put it in Context.qml along with other properties/settings so it can be found easily.
The property's name should be in lower case, so it doesn't collide with widget names, if any.

This is the "Search component": https://github.com/Dax89/harbour-sailorgram/blob/master/harbour-sailorgram/qml/components/search/MessageSearchList.qml

Only the component named "searchbox" needs to be hidden.
I order to start Sailorgram with search box not visible it's sufficient to assign its "visible" property with "showSearchField".

The "context" property is already there, so it's pretty easy to do it.

Basically you need to do the following thing

MessageSearchList.qml

```
// The height property is calculated dynamically

height: {
      var h = lvsearch.contentHeight;

      if(searchbox.visible)
          h += searchbox.height;

      return h;
}

// And the searchbox properties are binded dynamically as well (without states):

SearchField
{
   id: searchbox
   anchors { left: parent.left; right: parent.right }
   visible: context.showsearchfield
   height: visible ? Theme.itemSizeSmall : 0

   onVisibleChanged: {
       if(visible)
            return;

      searchbox.text = "";
     }
}
```

The pulldownmenu is ok :)
And, there is no need to change the anchors, the height + visible property are sufficient.

**Comment by Dax89** [comment]
Oh, if something is not clear, I'm here to explain everything!

**Comment by mentaljam** [comment]
I started with the "Search component" and did not managed as I chose the wrong way. I have a question: if I move the "showsearchsield" property to the "context", how do I properly connect to its changed signal? Is this right:

``` qml
    Component.onCompleted: {
        context.onShowsearchfieldChanged.connect(setSearchfieldItemText);
    }

    function setSearchfieldItemText() {
        searchfield.text = context.showsearchfield ? qsTr("Hide Search Field") : qsTr("Show Search Field");
    }
```

**Comment by Dax89** [comment]
In order to change the menu's text? :)

In qml every item's property is updated automatically everytime one (or more) assigned property changes, so you need to do only:

```
MenuItem
{
     text: context.showsearchfield ? qsTr("Hide Search Field") : qsTr("Show Search Field");
}
```

When "showsearchfield" changes to true/false the message is updated automatically :)

**Comment by mentaljam** [comment]
Ok, I should read more docs before asking))

**Comment by Dax89** [comment]
No problem! :+1:
